packages/data-stores/src/automated-transfer-eligibility/selectors.ts

```diff
@@ -18,77 +16,58 @@ export const getAutomatedTransferEligibility = (
 	return state[ siteId ];
 };
 
-export const getEligibilityHolds = createRegistrySelector(
-	( select: TransferSelectFn ) =>
-		( state: State, siteId: number | null ): TransferEligibilityError[] | null => {
-			const transferEligibility = select( STORE_KEY ).getAutomatedTransferEligibility( siteId );
-
-			if ( ! transferEligibility ) {
-				return null;
-			}
-
-			const { errors } = transferEligibility;
+export const getEligibilityHolds = (
+	state: State,
+	siteId: number | null
+): TransferEligibilityHold[] | null => {
+	const transferEligibility = getAutomatedTransferEligibility( state, siteId );
 
-			if ( ! errors ) {
-				return null;
-			}
+	if ( ! transferEligibility?.errors ) {
+		return null;
+	}
 
-			return errors.map( ( { code } ) => {
-				return statusMapping[ code ] ?? '';
-			} );
+	return transferEligibility.errors.reduce< TransferEligibilityHold[] >( ( errs, { code } ) => {
+		if ( code in statusMapping ) {
+			// At this point, we know code is a key of statusMapping, so we help TS understand that.
+			errs.push( statusMapping[ code as keyof typeof statusMapping ] );
 		}
-);
-
-export const getEligibilityWarnings = createRegistrySelector(
-	( select: TransferSelectFn ) =>
-		( state: State, siteId: number | null ): TransferEligibilityWarning[] | null => {
-			const transferEligibility = select( STORE_KEY ).getAutomatedTransferEligibility( siteId );
-
-			if ( ! transferEligibility ) {
-				return null;
-			}
-
-			const { warnings } = transferEligibility;
-
-			if ( ! warnings ) {
-				return null;
-			}
-
-			// combine plugin and theme warnings into one list
-			const combined = Object.values( warnings ).flat();
+		return errs;
+	}, [] );
```

Comment:
This function was completely incorrect before, and I wonder how many bugs it caused. Previously, the function was _actually_ returning an array of strings, while claiming that it returned `TransferEligibilityError[]`, which is an object type with the shape `{ code, message }`. And the one place that consumed it interacted with this array as if it contained objects. 

**In other words, this selector was previously not doing anything at all.**

Since this is only consumed in a single place, and everywhere that consumes it only needed to interact with the `code`, I changed the type to be much more specific. It's now restricted to values of the statusMapping object. I also used `reduce` to avoid putting empty strings in the array.

So while it looks like a bigger change, this is basically just correctly encoding the previous behavior -- but with correct and specific types now.

Above, we updated the consumer of this selector to handle the types correctly.

Reply:
Nice catch, @noahtallen!